pkg/build/builder/docker.go

```diff
@@ -53,9 +56,15 @@ func (d *DockerBuilder) Build() error {
 	if err != nil {
 		return err
 	}
-	if err := fetchSource(buildDir, d.build, d.urlTimeout, os.Stdin, d.git); err != nil {
+	sourceInfo, err := fetchSource(buildDir, d.build, d.urlTimeout, os.Stdin, d.git)
+	if err != nil {
 		return err
 	}
+	if sourceInfo != nil {
+		if err := updateBuildRevision(d.client, d.build, sourceInfo); err != nil {
```

Comment:
This breaks backwards compatibility for your image.  Old masters won't be able to run new builder images, right?  Any way you can reasonably tolerate this failure?